src/_ert/events.py

```diff
@@ -0,0 +1,240 @@
+
+
+class ForwardModelStepBaseEvent(BaseEvent):
+    ensemble: Union[str, None] = None
+    real: str
+    fm_step: str
+
+
+class ForwardModelStepStart(ForwardModelStepBaseEvent):
+    event_type: Id.FORWARD_MODEL_STEP_START_TYPE = Id.FORWARD_MODEL_STEP_START
+    std_out: Union[str, None] = None
+    std_err: Union[str, None] = None
+
+
+class ForwardModelStepRunning(ForwardModelStepBaseEvent):
+    event_type: Id.FORWARD_MODEL_STEP_RUNNING_TYPE = Id.FORWARD_MODEL_STEP_RUNNING
+    max_memory_usage: Union[int, None] = None
+    current_memory_usage: Union[int, None] = None
+
+
+class ForwardModelStepSuccess(ForwardModelStepBaseEvent):
+    event_type: Id.FORWARD_MODEL_STEP_SUCCESS_TYPE = Id.FORWARD_MODEL_STEP_SUCCESS
+    current_memory_usage: Union[int, None] = None
+
+
+class ForwardModelStepFailure(ForwardModelStepBaseEvent):
+    event_type: Id.FORWARD_MODEL_STEP_FAILURE_TYPE = Id.FORWARD_MODEL_STEP_FAILURE
+    error_msg: str
+    exit_code: Union[int, None] = None
+
+
+class ForwardModelStepChecksum(BaseEvent):
+    event_type: Id.FORWARD_MODEL_STEP_CHECKSUM_TYPE = Id.FORWARD_MODEL_STEP_CHECKSUM
+    ensemble: Union[str, None] = None
+    real: str
+    checksums: Dict[str, Dict[str, Any]]
+
+
+class RealizationBaseEvent(BaseEvent):
+    real: str
+    ensemble: Union[str, None] = None
+    queue_event_type: Union[str, None] = None
+
+
+class RealizationPending(RealizationBaseEvent):
+    event_type: Id.REALIZATION_PENDING_TYPE = Id.REALIZATION_PENDING
+
+
+class RealizationRunning(RealizationBaseEvent):
+    event_type: Id.REALIZATION_RUNNING_TYPE = Id.REALIZATION_RUNNING
+
+
+class RealizationSuccess(RealizationBaseEvent):
+    event_type: Id.REALIZATION_SUCCESS_TYPE = Id.REALIZATION_SUCCESS
+
+
+class RealizationFailed(RealizationBaseEvent):
+    event_type: Id.REALIZATION_FAILURE_TYPE = Id.REALIZATION_FAILURE
+    callback_status_message: Union[str, None] = None  # Only used for JobState.FAILED
+
+
+class RealizationUnknown(RealizationBaseEvent):
+    event_type: Id.REALIZATION_UNKNOWN_TYPE = Id.REALIZATION_UNKNOWN
+
+
+class RealizationWaiting(RealizationBaseEvent):
+    event_type: Id.REALIZATION_WAITING_TYPE = Id.REALIZATION_WAITING
+
+
+class RealizationTimeout(RealizationBaseEvent):
+    event_type: Id.REALIZATION_TIMEOUT_TYPE = Id.REALIZATION_TIMEOUT
+
+
+class EnsembleBaseEvent(BaseEvent):
+    ensemble: Union[str, None] = None
+
+
+class EnsembleStarted(EnsembleBaseEvent):
+    event_type: Id.ENSEMBLE_STARTED_TYPE = Id.ENSEMBLE_STARTED
+
+
+class EnsembleSucceeded(EnsembleBaseEvent):
+    event_type: Id.ENSEMBLE_SUCCEEDED_TYPE = Id.ENSEMBLE_SUCCEEDED
+
+
+class EnsembleFailed(EnsembleBaseEvent):
+    event_type: Id.ENSEMBLE_FAILED_TYPE = Id.ENSEMBLE_FAILED
+
+
+class EnsembleCancelled(EnsembleBaseEvent):
+    event_type: Id.ENSEMBLE_CANCELLED_TYPE = Id.ENSEMBLE_CANCELLED
+
+
+class EESnapshot(BaseEvent):
+    event_type: Id.EE_SNAPSHOT_TYPE = Id.EE_SNAPSHOT
+    snapshot: Any
+
+
+class EESnapshotUpdate(BaseEvent):
```

Comment:
Prefix EE means that the event is created by the evaluator?

Reply:
EE is EnsembleEvaluator, yes. Did not change it from the name it had before this PR

Reply:
Hm, what do you mean? There were no EE* events to my best knowledge at least.

Reply:
exactly :)